presto-memory/src/test/java/io/prestosql/plugin/memory/TestMemorySmoke.java

```diff
@@ -263,6 +264,61 @@ public void testSemiJoinDynamicFilteringBlockProbeSide()
                 ImmutableSet.of(1, ORDERS_COUNT, PART_COUNT));
     }
 
+    @Test
+    public void testCrossJoinDynamicFiltering()
+    {
+        assertUpdate("CREATE TABLE probe (k VARCHAR, v INTEGER)");
+        assertUpdate("CREATE TABLE build (vmin INTEGER, vmax INTEGER)");
+        assertUpdate("INSERT INTO probe VALUES ('a', 0), ('b', 1), ('c', 2), ('d', 3)", 4);
+        assertUpdate("INSERT INTO build VALUES (1, 2)", 1);
+
+        assertDynamicFiltering("SELECT * FROM probe JOIN build ON v >= vmin", withBroadcastJoin(), 3, ImmutableSet.of(3, 1));
+        assertDynamicFiltering("SELECT * FROM probe JOIN build ON v > vmin", withBroadcastJoin(), 2, ImmutableSet.of(2, 1));
+        assertDynamicFiltering("SELECT * FROM probe JOIN build ON v <= vmax", withBroadcastJoin(), 3, ImmutableSet.of(3, 1));
+        assertDynamicFiltering("SELECT * FROM probe JOIN build ON v < vmax", withBroadcastJoin(), 2, ImmutableSet.of(2, 1));
+
+        assertDynamicFiltering("SELECT * FROM probe JOIN build ON v >= vmin AND v < vmax", withBroadcastJoin(), 1, ImmutableSet.of(1));
```

Comment:
why `ImmutableSet.of(1)` has one element here?

Reply:
You're right, it's better to use `ImmutableList.of(1, 1)` here.